lib/paperclip/io_adapters/http_url_proxy_adapter.rb

```diff
@@ -9,8 +11,8 @@ def self.register
     REGEXP = /\Ahttps?:\/\//.freeze
 
     def initialize(target, options = {})
-      escaped = URI.escape(target)
-      super(URI(target == URI.unescape(target) ? escaped : target), options)
+      escaped = Addressable::URI.escape(target)
```

Comment:
@YuheiNakasaka 

Using URI encode_www_form_component might be a better solution since they replaced .escape with it. 
In this case, 
URI.encode_www_form_component(target).gsub('+', '%20')